src/common/dns_utils.cpp

```diff
@@ -327,10 +299,14 @@ std::vector<std::string> DNSResolver::get_record(const std::string& url, int rec
   std::vector<std::string> addresses;
   dnssec_available = false;
   dnssec_valid = false;
-
-  // destructor takes care of cleanup
-  ub_result_ptr result;
-
+  
+  ub_result *result;
+  // Make sure we are cleaning after result.
+  epee::misc_utils::auto_scope_leave_caller scope_exit_handler =
+    epee::misc_utils::create_scope_leave_handler([&](){
+    ub_resolve_free(result);
```

Comment:
Should a `nullptr` check be added here?

Reply:
I guess it always worked before, so probably not.

Reply:
Good catch @vtnerd . 

To be honest, I thought about adding a simple nullptr check, but I didn't want to make any semantical changes to the code.

Reply:
I am resolving this conversation. Please feel free to reopen this comment.